### PR TITLE
Rewrites incorrect documentation for the pickweight() proc

### PR DESCRIPTION
**Pull request by ghost**
Whoever documented `pickweight` is a butt and put down incorrect information. Now how the elements are picked is properly documented and I can properly sleep again.

### CODE REVIEW

**Comment by RemieRichards** [comment]
Denied, bad proc who's job is already taken by an alt syntax of ```prob()```, let it stay referenced as dumb

**Comment by optimumtact** [comment]
We shouldn't encourage bad information, if you don't like the proc, ask them to remove it instead.

**Comment by RemieRichards** [comment]
Xhuis' documentation is no improvement.
It adds nothing and simply repeats the steps the code follows... WHICH IS THE CODE.

**Comment by Xhuis** [comment]
The earlier comment is both incorrect and misleading. The new one is at the very least accurate and explains what the code does instead of actively serving to mislead whoever's reading it. If you can provide a convincing argument for why the earlier one should stay, then I'd be happy to hear it.

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -177,7 +177,11 @@
 		result = first ^ second
 	return result
 
-//Pretends to pick an element based on its weight but really just seems to pick a random element.
+//Picks a random element from a list based on a weighting system:
+//1. Adds up the total of weights for each element
+//2. Gets a number between 1 and that total
+//3. For each element in the list, subtracts its weighting from that number
+//4. If that makes the number 0 or less, return that element.
```
Comment (ChangelingRain) [inline code/__HELPERS/_lists.dm]: it's not a useful comment if you can easily see what the code does by looking at it.

**Comment by RemieRichards** [comment]
Because ```pickweight``` should be deprecated, it was either:
1. Added by someone who didn't know about ```pick("A":45, "B":60)```
or
2. Added BEFORE the pick syntax.

The entire proc should go, here's the steps to progress, where we are now, and where we'd be with your PR.

```
-1 ----------->---------- 0 ----------->---------- 1
Your PR                Current               No pickweight()
```

File: code/__HELPERS/_lists.dm (hunk 1 above)
Comment (Xhuis) [inline code/__HELPERS/_lists.dm]: It's also not useful when it actively _misleads_ the person reading it.

**Comment by Xhuis** [comment]
Why did you not say that in the first place? If you had asked me to just replace pickweight instead of telling me the comment was bad, wouldn't that have avoided unnecessary conflict to begin with?

**Comment by RemieRichards** [comment]
>"If I'd have listened to you the first time, things wouldn't be shit!"

**Comment by Qustinnus** [comment]
what even is the point of this argument?

**Comment by Xhuis** [comment]
You're probably right in that there isn't one and that we're both being needlessly argumentative. I'm going to try to solve the issue by just removing `pickweight` so we can stop being so upset over silly code comments in a 2D atmos sim.

**Comment by skull132** [comment]
> Also, to pick a value randomly from a list, simply pass the list as the sole argument into pick(). In this case, there is no provision for weighting the probabilities. All items in the list are equally likely.

🤔

**Comment by ChangelingRain** [comment]
pickweight does have a use, it's just that that use is exclusively to pick from an already-existing associated list that you plan to modify.

**Comment by Xhuis** [comment]
I don't see a reason to remove it, then. I can try to use byond's functionality with pickweight as a wrapper, however.

**Comment by MrPerson** [comment]
BYOND's weird pick() syntax is nonstandard and really annoying to work with. Lummox wants to rewrite it anyway, why bother using it? pickweight is fine and useful.

**Pull request merged**